03_letters/shahriys/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var outletter io.Writer = os.Stdout
+
+func letters(s string) map[rune]int {
+
+	alphamap := make(map[rune]int)
+
+	for i := 0; i < len(s); i++ {
+		alphamap[rune(s[i])]++
+	}
+	//fmt.Println(alphamap)
+	return alphamap
+}
+
+// As rune type can't be sorted so creating new type
+type RuneSlice []rune
+
+func (p RuneSlice) Len() int           { return len(p) }
+func (p RuneSlice) Less(i, j int) bool { return p[i] < p[j] }
+func (p RuneSlice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+func sortLetters(m map[rune]int) []string {
+
+	var keys = make([]rune, 0, len(m))
+	for key := range m {
+		keys = append(keys, key)
+	}
+	var runes RuneSlice = keys
```

Comment:
You can avoid having to create a custom type which implements the Sortable interface if you avoid creating a `[]rune` from the map. Is there a way in which you could directly make a `[]string`? You are doing it below... Are you able to generate that from the map directly, without the step inbetween?